cli/crates/gateway/src/executor.rs

```diff
@@ -168,10 +167,21 @@ impl gateway_core::Executor for Executor {
     ) -> Result<Self::Response, crate::Error> {
         let schema = self.build_schema(&ctx, auth).await?;
         let payload_stream = Box::pin(schema.execute_stream(request));
+        let mut response_builder = axum::response::Response::builder();
         let (headers, bytes_stream) =
             gateway_core::encode_stream_response(ctx.as_ref(), payload_stream, streaming_format).await;
-        Ok((headers, axum::body::StreamBody::new(bytes_stream))
-            .into_response()
+        response_builder
+            .headers_mut()
+            .unwrap()
```

Comment:
Would it be possible to either add an error here or an expect depending on whether it's supposed to be fallible? Same for the other unwrap